### PR TITLE
feat: Add setting for minimal degrees search distance for weather station

### PR DESCRIPTION
**Pull request by Ahmad-Wahid**
The user has the capability to modify the degree search distance  for  weather stations in the `flexmeasures.cfg` file by assigning a new value to the parameter `MINIMAL_WEATHER_STATIONS`. The default value for this parameter is set to 2.

Input on naming improvement would be appreciated.

### CODE REVIEW

File: flexmeasures_openweathermap/__init__.py [hunk 1]
```diff
@@ -48,6 +49,10 @@
         description=f"Name of the weather station asset, defaults to '{DEFAULT_WEATHER_STATION_NAME}'",
         level="debug",
     ),
+    "MINIMAL_WEATHER_STATIONS": dict(
```
Comment (nhoening) [inline flexmeasures_openweathermap/__init__.py:52]: Code looks good, but looking at the setting name and description, you misunderstood what the setting does. It limits the geographical search radius (in degree latitude as well as longitude) in which weather stations need to be to be useful for storing the forecasts.
We always use the nearest, but if that is too far away, we rather raise.

**Review by nhoening — CHANGES_REQUESTED** [review]

File: flexmeasures_openweathermap/__init__.py (hunk 1 above)
Comment (Ahmad-Wahid) [reply flexmeasures_openweathermap/__init__.py]: I thought of nearest station based on these degrees. Thank you.

File: flexmeasures_openweathermap/__init__.py [hunk 2]
```diff
@@ -48,6 +49,10 @@
         description=f"Name of the weather station asset, defaults to '{DEFAULT_WEATHER_STATION_NAME}'",
         level="debug",
     ),
+    "MINIMAL_DISTANCE_DEGREE": dict(
+        descripion=f"Number of the minimum distance degree for weather station, defaults to '{DEFAULT_MINIMAL_DISTANCE_DEGREE}'",
```
Comment (nhoening) [inline flexmeasures_openweathermap/__init__.py:53]: I think the english is a bit clunky, let's use this:

"Maximum distance (in degrees latitude & longitude) for weather stations from forecast location, defaults to {DEFAULT_MAXIMAL_DEGREE_LOCATION_DISTANCE}"

File: flexmeasures_openweathermap/__init__.py [hunk 3]
```diff
@@ -48,6 +49,10 @@
         description=f"Name of the weather station asset, defaults to '{DEFAULT_WEATHER_STATION_NAME}'",
         level="debug",
     ),
+    "MINIMAL_DISTANCE_DEGREE": dict(
```
Comment (nhoening) [inline flexmeasures_openweathermap/__init__.py:52]: Not sure why you switched maximal to minimal :-)
And I feel we should prefix with "OPENWEATHERMAP_", so it's clear in the config file where this belongs to.

Let's call it `OPENWEATHERMAP_MAXIMAL_DEGREE_LOCATION_DISTANCE`

(the setting `WEATHER_STATION_NAME` also should have this prefix, but let's not deal with that now)

**Review by nhoening — CHANGES_REQUESTED** [review]

**Comment by nhoening** [comment]
I also think that 1 is a better default.

**Review by nhoening — APPROVED** [review]

**Pull request merged**